pyomo/contrib/mindtpy/iterate.py

```diff
@@ -163,3 +279,75 @@ def algorithm_should_terminate(solve_data, config, check_cycling):
     #         'Exiting iteration loop.')
     #     return True
     return False
+
+
+def bound_fix(solve_data, config, last_iter_cuts):
+    if config.single_tree:
+        config.logger.info(
+            'Fix the bound to the value of one iteration before optimal solution is found.')
+        if solve_data.results.problem.sense == ProblemSense.minimize:
+            solve_data.LB = solve_data.stored_bound[solve_data.UB]
+        else:
+            solve_data.UB = solve_data.stored_bound[solve_data.LB]
+    else:
+        config.logger.info(
+            'Solve the master problem without the last nogood cut to fix the bound.'
+            'zero_tolerance is set to 1E-4')
+        config.zero_tolerance = 1E-4
+        # Solve NLP subproblem
+        # The constraint linearization happens in the handlers
+        if last_iter_cuts is False:
+            fixed_nlp, fixed_nlp_result = solve_NLP_subproblem(
+                solve_data, config)
+            if fixed_nlp_result.solver.termination_condition in {tc.optimal, tc.locallyOptimal, tc.feasible}:
+                handle_NLP_subproblem_optimal(fixed_nlp, solve_data, config)
+            elif fixed_nlp_result.solver.termination_condition is tc.infeasible:
+                handle_NLP_subproblem_infeasible(fixed_nlp, solve_data, config)
+            else:
+                handle_NLP_subproblem_other_termination(fixed_nlp, fixed_nlp_result.solver.termination_condition,
+                                                        solve_data, config)
+
+        MindtPy = solve_data.mip.MindtPy_utils
+        # TODO: only deactivate the last integer cut.
```

Comment:
This is no longer the case. We actually deactivate all the nogood cuts that appeared since the UB was found (now we can be sure we are rigorous with the LB).